pkg/apply/event/event.go

```diff
@@ -158,72 +158,71 @@ func (ee ErrorEvent) String() string {
 	return fmt.Sprintf("ErrorEvent{ Err: %q }", ee.Err.Error())
 }
 
-//go:generate stringer -type=WaitEventOperation -linecomment
-type WaitEventOperation int
+//go:generate stringer -type=WaitEventStatus -linecomment
+type WaitEventStatus int
 
 const (
-	ReconcilePending WaitEventOperation = iota // Pending
-	Reconciled
-	ReconcileSkipped // Skipped
-	ReconcileTimeout // Timeout
-	ReconcileFailed  // Failed
+	ReconcilePending    WaitEventStatus = iota // Pending
+	ReconcileSuccessful                        // Successful
+	ReconcileSkipped                           // Skipped
+	ReconcileTimeout                           // Timeout
+	ReconcileFailed                            // Failed
 )
 
 type WaitEvent struct {
 	GroupName  string
 	Identifier object.ObjMetadata
-	Operation  WaitEventOperation
+	Status     WaitEventStatus
 }
 
 // String returns a string suitable for logging
 func (we WaitEvent) String() string {
 	return fmt.Sprintf("WaitEvent{ GroupName: %q, Operation: %q, Identifier: %q }",
-		we.GroupName, we.Operation, we.Identifier)
+		we.GroupName, we.Status, we.Identifier)
 }
 
-//go:generate stringer -type=ActionGroupEventType
-type ActionGroupEventType int
+//go:generate stringer -type=ActionGroupEventStatus
+type ActionGroupEventStatus int
 
 const (
-	Started ActionGroupEventType = iota
+	Started ActionGroupEventStatus = iota
 	Finished
 )
 
 type ActionGroupEvent struct {
 	GroupName string
 	Action    ResourceAction
-	Type      ActionGroupEventType
+	Status    ActionGroupEventStatus
 }
 
 // String returns a string suitable for logging
 func (age ActionGroupEvent) String() string {
 	return fmt.Sprintf("ActionGroupEvent{ GroupName: %q, Action: %q, Type: %q }",
-		age.GroupName, age.Action, age.Type)
+		age.GroupName, age.Action, age.Status)
 }
 
-//go:generate stringer -type=ApplyEventOperation
-type ApplyEventOperation int
+//go:generate stringer -type=ApplyEventStatus -linecomment
+type ApplyEventStatus int
 
 const (
-	ApplyUnspecified ApplyEventOperation = iota
-	ServersideApplied
-	Created
-	Unchanged
-	Configured
+	ApplyPending    ApplyEventStatus = iota // Pending
+	ApplySuccessful                         // Successful
```

Comment:
Do we need to preserve the distinction between Client-side Apply (Updated and Created) and Server-side Apply in the ApplyEvent?

If so, can it be done with a different field, like `ApplyMode: ServerSide|ClientSide`? 
Or do we also need to distinguish between Created, Updated, and Unchanged?

I think in the context of Config Sync, any time there's a retry the new Created/Updated will just be replaced by Unchanged, because the live state will already be as desired. So it doesn't really matter there. But maybe it matters for kpt, which doesn't retry automatically?

Reply:
I don't think we need to distinguish between SSA and client-side apply in the event. We use whatever approach the user provided as the argument (with one special case I think).

Client-side apply currently does tell us whether a resource was created, updated or unchanged, which is kinda useful. I know SSA doesn't distinguish between updated and unchanged, but does it tell us if a resources was created rather than updated/unchanged?

Reply:
> Client-side apply currently does tell us whether a resource was created, updated or unchanged, which is kinda useful.

Yeah, my feeling was that if we need this we can add another field to clarify it. That actually IS an "CRUD operation" and not just a "status". So we could have both operation & status just for apply events, if you think it's needed. 

I don't think anyone is actually using it from the JSON output today (of the clients we know about), and it wasn't exposed through Config Sync until 1.11.0 added it to the ResourceGroup object status. So I doubt anyone will complain. And we can always add it back later.

> I know SSA doesn't distinguish between updated and unchanged, but does it tell us if a resources was created rather than updated/unchanged?

Even if the API exposes it, the kubectl library we're using doesn't expose it. That's where these "operations" initially come from.